src/controller/SetUpCodePairer.cpp

```diff
@@ -89,10 +93,16 @@ CHIP_ERROR SetUpCodePairer::PairDevice(NodeId remoteId, const char * setUpCode,
         return CHIP_NO_ERROR;
     }
 
-    ReturnErrorOnFailure(Connect(payload));
-    return mSystemLayer->StartTimer(System::Clock::Milliseconds32(kDeviceDiscoveredTimeout), OnDeviceDiscoveredTimeoutCallback,
-                                    this);
+    ReturnErrorOnFailureWithMetric(kMetricSetupCodePairerPairDevice, Connect(payload));
+    auto errorCode =
+        mSystemLayer->StartTimer(System::Clock::Milliseconds32(kDeviceDiscoveredTimeout), OnDeviceDiscoveredTimeoutCallback, this);
+    if (CHIP_NO_ERROR == errorCode)
+    {
+        MATTER_LOG_METRIC_BEGIN(kMetricSetupCodePairerPairDevice);
```

Comment:
Again, I don't quite understand how MATTER_LOG_METRIC_BEGIN and ReturnErrorOnFailureWithMetric work when using the same metric name....

Reply:
The begin and end are used to make longer sync stages. The ReturnErrorOnFailureWithMetric is used for any errors that happen before the begin is marked. If there are errors, they are just marked with the same even name but not begin/end modes

Reply:
Yup